Build failure: godot_4-mono

### Steps To Reproduce

Steps to reproduce the behavior:
1. Build `godot_4-mono`

### Build log

```
    inlined from 'viaByteSinkToCharString' at thirdparty/icu4c/common/bytesinkutil.h:147:5,
    inlined from 'ulocimp_canonicalize_75_godot' at thirdparty/icu4c/common/uloc.cpp:2145:12:
thirdparty/icu4c/common/cmemory.cpp:99:13: warning: 'free' called on unallocated object 'result' [8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wfree-nonheap-object-Wfree-nonheap-object8;;]
   99 |             uprv_default_free(buffer);
      |             ^
thirdparty/icu4c/common/bytesinkutil.h: In function 'ulocimp_canonicalize_75_godot':
thirdparty/icu4c/common/bytesinkutil.h:143:20: note: declared here
  143 |         CharString result;
      |                    ^
In function 'uprv_free_75_godot',
    inlined from 'releaseArray' at thirdparty/icu4c/common/cmemory.h:460:22,
    inlined from '__dt_base ' at thirdparty/icu4c/common/cmemory.h:363:38,
    inlined from '__dt_base ' at thirdparty/icu4c/common/charstr.h:56:20,
    inlined from 'getKeywordValue' at thirdparty/icu4c/common/unicode/locid.h:1267:1,
    inlined from '_copyExtensions' at thirdparty/icu4c/common/localebuilder.cpp:186:69,
    inlined from 'build' at thirdparty/icu4c/common/localebuilder.cpp:463:24:
thirdparty/icu4c/common/cmemory.cpp:99:13: warning: 'free' called on unallocated object 'result' [8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wfree-nonheap-object-Wfree-nonheap-object8;;]
   99 |             uprv_default_free(buffer);
      |             ^
thirdparty/icu4c/common/unicode/locid.h: In member function 'build':
thirdparty/icu4c/common/unicode/locid.h:1263:17: note: declared here
 1263 |     StringClass result;
      |                 ^
thirdparty/freetype/src/autofit/afglobal.c: In function 'af_face_globals_new':
thirdparty/freetype/src/autofit/afglobal.c:344:5: warning: 'memset' writing 696 bytes into a region of size 0 overflows the destination [8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wstringop-overflow=-Wstringop-overflow=8;;]
  344 |     FT_ZERO( &globals->metrics );
      |     ^
lto1: note: destination object is likely at address zero
servers/audio/effects/audio_effect_pitch_shift.cpp: In member function 'PitchShift.constprop':
servers/audio/effects/audio_effect_pitch_shift.cpp:163:31: warning: 'memset' specified bound between 18446744065119617024 and 18446744073709551608 exceeds maximum object size 9223372036854775807 [8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wstringop-overflow=-Wstringop-overflow=8;;]
  163 |                         memset(gSynMagn, 0, fftFrameSize*sizeof(float));
      |                               ^
servers/audio/effects/audio_effect_pitch_shift.cpp:164:31: warning: 'memset' specified bound between 18446744065119617024 and 18446744073709551608 exceeds maximum object size 9223372036854775807 [8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wstringop-overflow=-Wstringop-overflow=8;;]
  164 |                         memset(gSynFreq, 0, fftFrameSize*sizeof(float));
      |                               ^
lto1: fatal error: error writing to /tmp/ccNmqlp6.s: No space left on device
compilation terminated.
make: *** [/tmp/ccbN58pm.mk:338: /tmp/ccLYWg5c.ltrans112.ltrans.o] Error 1
make: *** Waiting for unfinished jobs....
lto1: fatal error: error writing to /tmp/ccBDcesr.s: No space left on device
compilation terminated.
make: *** [/tmp/ccbN58pm.mk:335: /tmp/ccLYWg5c.ltrans111.ltrans.o] Error 1
lto-wrapper: fatal error: make returned 2 exit status
compilation terminated.
/nix/store/pbixrqd120nnw8a0ygfpm3mnwlr1mbj7-binutils-2.43.1/bin/ld: error: lto-wrapper failed
collect2: error: ld returned 1 exit status
scons: *** [bin/godot.linuxbsd.editor.x86_64.mono] Error 1
scons: building terminated because of errors.
[Time elapsed: 00:48:45.12]
```

### Additional context

You can check the failing Hydra job here: https://hydra.nixos.org/build/278099603

### Notify maintainers

<!--
Please @ people who are in the `meta.maintainers` list of the offending package or module.
If in doubt, check `git blame` for whoever last touched something.
-->

@shiryel @corngood

### Metadata

<!-- Please insert the output of running `nix-shell -p nix-info --run "nix-info -m"` below this line -->
 - system: `"x86_64-linux"`
 - host os: `Linux 6.6.59, NixOS, 24.11 (Vicuna), 24.11.20241109.76612b1`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.24.10`
 - nixpkgs: `/nix/store/sh0v21n9zc03b94qjj4jkkg5mbsjkpzr-source`

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/by-name/go/godot_4/package.nix
+++ b/pkgs/by-name/go/godot_4/package.nix
@@ -231,6 +231,11 @@ let
       };
     };
 
+    requiredSystemFeatures = [
+      # fixes: No space left on device
+      "big-parallel"
+    ];
+
     meta = {
       changelog = "https://github.com/godotengine/godot/releases/tag/${version}";
       description = "Free and Open Source 2D and 3D game engine";

--- a/pkgs/by-name/go/godot_4/package.nix
+++ b/pkgs/by-name/go/godot_4/package.nix
@@ -26,6 +26,7 @@
   scons,
   speechd-minimal,
   stdenv,
+  testers,
   udev,
   vulkan-loader,
   wayland,
@@ -58,7 +59,7 @@ let
 
   arch = stdenv.hostPlatform.linuxArch;
 
-  attrs = rec {
+  attrs = finalAttrs: rec {
     pname = "godot4${suffix}";
     version = "4.3-stable";
     commitHash = "77dcf97d82cbfe4e4615475fa52ca03da645dbd8";
@@ -223,6 +224,13 @@ let
         runHook postInstall
       '';
 
+    passthru.tests = {
+      version = testers.testVersion {
+        package = finalAttrs.finalPackage;
+        version = lib.replaceStrings [ "-" ] [ "." ] version;
+      };
+    };
+
     meta = {
       changelog = "https://github.com/godotengine/godot/releases/tag/${version}";
       description = "Free and Open Source 2D and 3D game engine";
@@ -242,7 +250,6 @@ let
 
 in
 stdenv.mkDerivation (
-  finalAttrs:
   if withMono then
     dotnetCorePackages.addNuGetDeps {
       nugetDeps = ./deps.nix;
@@ -252,7 +259,7 @@ stdenv.mkDerivation (
           old.buildInputs
           ++ lib.concatLists (lib.attrValues (lib.getAttrs runtimeIds dotnet-sdk_8.targetPackages));
       };
-    } attrs finalAttrs
+    } attrs
   else
     attrs
 )

--- a/pkgs/by-name/go/godot_4/package.nix
+++ b/pkgs/by-name/go/godot_4/package.nix
@@ -56,6 +56,8 @@ let
 
   suffix = if withMono then "-mono" else "";
 
+  arch = stdenv.hostPlatform.linuxArch;
+
   attrs = rec {
     pname = "godot4${suffix}";
     version = "4.3-stable";
@@ -149,9 +151,9 @@ let
     postBuild = lib.optionalString withMono ''
       echo "Generating Glue"
       if [[ ${withPrecision} == *double* ]]; then
-          bin/godot.${withPlatform}.${withTarget}.${withPrecision}.x86_64.mono --headless --generate-mono-glue modules/mono/glue
+          bin/godot.${withPlatform}.${withTarget}.${withPrecision}.${arch}.mono --headless --generate-mono-glue modules/mono/glue
       else
-          bin/godot.${withPlatform}.${withTarget}.x86_64.mono --headless --generate-mono-glue modules/mono/glue
+          bin/godot.${withPlatform}.${withTarget}.${arch}.mono --headless --generate-mono-glue modules/mono/glue
       fi
       echo "Building C#/.NET Assemblies"
       python modules/mono/build_scripts/build_assemblies.py --godot-output-dir bin --precision=${withPrecision}

